### ISSUE TITLE
[Bug] echarts lines effect 5.3版本显示正常，5.4开始循环时运动结束会闪烁一下

### ISSUE THREAD

**Issue by Rookior** [apache/echarts#20216]
### Version

5.4

### Link to Minimal Reproduction

https://echarts.apache.org/examples/zh/editor.html?c=calendar-graph&version=5.5.1&theme=dark

### Steps to Reproduce

option = {
  xAxis: {
    type: 'value',
    min: 0,
    max: 100,
    axisLine: { show: false },
    axisTick: { show: false },
    splitLine: { show: false }
  },
  yAxis: {
    type: 'value',
    min: 0,
    max: 100,
    inverse: true,
    axisLine: { show: false },
    axisTick: { show: false },
    splitLine: { show: false }
  },
  series: [
    {
      type: 'lines',
      name: '陕西-北京',
      zlevel: 2,
      symbol: ['none', 'none'],
      symbolSize: 10,
      effect: {
        show: true,
        period: 6,
        trailLength: 0,
        symbol: 'arrow',
        color: '#FFC12F',
        symbolSize: 10
      },
      lineStyle: {
        normal: { color: '#fff', width: 2, opacity: 0.6, curveness: 0.2 }
      },
      coordinateSystem: 'cartesian2d',
      data: [
        {
          coords: [
            [54.1, 43.2],
            [58.8, 38]
          ]
        },
        {
          coords: [
            [54.1, 43.2],
            [58.8, 38]
          ]
        },
        {
          coords: [
            [54.1, 43.2],
            [58.8, 38]
          ]
        }
      ]
    }
  ]
};

### Current Behavior

5.3之前显示正常，5.4开始异常

### Expected Behavior

5.3之前显示正常，5.4开始异常

### Environment

```markdown
- OS:
- Browser:
- Framework:
```

### Any additional comments?

如何修改配置项使页面显示正常

**Comment by plainheart** [comment]
可能是 #16943 导致的？

**Comment by helgasoft** [comment]
yes, flicker probably introduced by [roundTrip](https://echarts.apache.org/en/option.html#series-lines.effect.roundTrip) from PR 16943. Arrow mirror image shows beyond end of the line for a split second. [Demo](https://echarts.apache.org/examples/en/editor.html?code=PYBwLglsB2AEC8sDeAoWsAeBBDEDOAXMmurGAJ4gCmRA5AG4CGANgK5W0A0J6AthNCIAGbqVi9GGIgEYhInrEn4AMgJrJYeABbAA7kQBmLPFVgBfUaSV4AKhADGAayJJNO_bCPMT5y-jwgzBBgqtDqrtp6hsamZiQWJOQ4-C4KFNR0TGwcfuICwrkSUrCy8mLWoeFuUZ4xvgrWdk4u1R5ePgliAUEhai2RbXVx6J2aVABOEFSEsADaCqhi6OnqtEFheFwK6NCMvKuAqmmA_pEAtIDoyoA1cltLsABezFT0VMxEAEy5_uS8AEbAL3O0aAwHKwQHA2gAXQ-sCoBgMVHsYFSN387iIYHG7Gh6Gok2AABMiABmbFkcaMCDMZRUaAAczAWgK21IeC-v3-tEY43GemuKPsf2A4zoAGIAGJigDC0leYr5N1ZPz-AGUILd1ESicz0AB6HU81jQfE2SYgWAEDHsZmjMTrKjKigPFq6CD4hlvTiweyscZPDYzIQAOle9RuAqF-IEjDA9vIeBjvDo9i5MbwEEY0Fe-PlpHx0cYRHmKOIxfQ4fG-JmRdL6Fmr3esAALETg1DtUtZgBWAAcge7nqJ3Yh7fQw-LNqWi1L5crhZHpDrDebrdJYi7vf7sEHY9LO5uE7EU-LM6r89r9c9y9ebZrtZ7fYHQ_ne6WwyWL7fw7MAG4gA&version=5.5.1&theme=dark)
Versions 5.3.0 and below are fine.
